protostar/testing/test_case_runners/test_case_runner.py

```diff
@@ -34,11 +38,18 @@ def __init__(
         self._output_recorder = output_recorder
         self._stopwatch = stopwatch
 
-    async def run(self) -> TestCaseResult:
+    async def run(self, state: TestExecutionState) -> TestCaseResult:
         try:
             with self._stopwatch.lap(self._test_case.test_fn_name):
                 execution_result = await self._run_test_case()
 
+            try:
+                assert_no_expected_calls(
+                    state.starknet.cheatable_state.cheatable_state.expected_contract_calls
+                )
+            except (ProtostarException, CheaterException) as e:
+                raise ReportedException from e
+
```

Comment:
I think this should be moved elsewhere, `test_case_runner` seems like a bad place to include cheating logic

Reply:
Yep, that seems like a good place for this logic